src/tests/compiler/ast/base/parameteredNodeTests.ts

```diff
@@ -178,4 +178,47 @@ describe(nameof(ParameteredNode), () => {
             doTest("function identifier(param, param2) {}", ["param", "param2"]);
         });
     });
+
+    describe(nameof<ParameteredNode>(m => m.convertParamsToDestructuredObject), () => {
+        it("should convert a function declaration's parameters", () => {
+            const { firstChild, sourceFile } = getInfoFromText<FunctionDeclaration>("function f(a: number, b: string[]) { }}");
+            firstChild.convertParamsToDestructuredObject();
+            expect(sourceFile.getText()).to.equal("function f({ a, b }: { a: number; b: string[]; }) { }}");
+        });
+        it("should only convert class' methods not overriding", () => {
+            const { sourceFile } = getInfoFromText<FunctionDeclaration>(`
+interface I {
+    m(a: Date[], b: boolean, c?: string, d: number[][] | undefined): string[]
+}
+class C implements I {
+    m(a: Date[], b: boolean, c?: string, d: number[][] | undefined) {
+        throw 1
+    }
+    n(a: Date[], b: boolean, c?: string, d: number[][] | undefined) {
+        throw 1
+    }
+}`.trim());
+            sourceFile.getClassOrThrow("C").getMethods().forEach(m => m.convertParamsToDestructuredObject());
+            expect(sourceFile.getText().trim()).to.equal(`
+interface I {
+    m(a: Date[], b: boolean, c?: string, d: number[][] | undefined): string[]
+}
+class C implements I {
+    m(a: Date[], b: boolean, c?: string, d: number[][] | undefined) {
+        throw 1
+    }
+    n({ a, b, c, d }: { a: Date[]; b: boolean; c?: string; d: number[][] | undefined; }) {
+        throw 1
+    }
+}`.trim());
+
+        });
+        it("should refactor all files", () => {
+            const { sourceFile, project } = getInfoFromText<FunctionDeclaration>("export function f(a: number, b: string[]) { }", {filePath: "f.ts"});
+            const sourceFile2 = project.createSourceFile("a.ts", `import {f} = require("./f"); f(1, "b");`);
+            sourceFile.getFunctionOrThrow("f").convertParamsToDestructuredObject();
+            expect(sourceFile.getText()).to.equal(`export function f({ a, b }: { a: number; b: string[]; }) { }`);
+            expect(sourceFile2.getText()).to.equal(`import {f} = require("./f"); f(1, "b");`);
```

Comment:
I'm having some trouble making it work for multiple files - this test is wrong... researching on typescript project...